staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/dispatcher.go

```diff
@@ -238,13 +240,184 @@ func (d *validatingDispatcher) Dispatch(ctx context.Context, attr admission.Attr
 	}
 	if len(errs) > 1 {
 		for i := 1; i < len(errs); i++ {
-			// TODO: merge status errors; until then, just return the first one.
 			utilruntime.HandleError(errs[i])
 		}
+		return aggregateWebhookErrors(errs)
 	}
 	return errs[0]
 }
 
+func aggregateWebhookErrors(errs []error) error {
+	if len(errs) == 1 {
+		return errs[0]
+	}
+
+	errorGroups := groupErrorsByReason(errs)
```

Comment:
(nit)

The "valueByKey" naming convention makes the key clear.

```suggestion
	errorsByReason := groupErrorsByReason(errs)
```

Reply:
Done.